Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM python:3.12-alpine
+LABEL maintainer="ohnooq1@gmail.com"
+
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONDONTWRITEBYTECODE=1
+
+WORKDIR /app
+
+COPY requirements.txt /app/
+RUN pip install --upgrade pip && pip install -r requirements.txt
```

Comment:
It's a good practice to separate the `pip install --upgrade pip` and `pip install -r requirements.txt` commands into different RUN instructions to leverage Docker's caching mechanism more effectively. This way, if only the requirements change, the pip upgrade step won't be repeated.